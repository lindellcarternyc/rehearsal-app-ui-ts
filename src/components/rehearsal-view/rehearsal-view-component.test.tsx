import * as React from 'react'
import * as ReactDOM from 'react-dom'
import * as renderer from 'react-test-renderer'

import RehearsalViewComponent, { RehearsalViewComponentProps } from './rehearsal-view-component'

const dismissRehearsal = jest.fn()

describe('RehearsalViewComponent', () => {
  it('renders without crashing', () => {
    const props: RehearsalViewComponentProps = {
      date: 'Tue, January 16',
      rehearsal: {
        time: '19:00 - 22:00',
        material: ['']
      },
      dismissRehearsal
    }
    const div = document.createElement('div')
    ReactDOM.render(
      <RehearsalViewComponent {...props} />,
      div
    )
  })

  it('renders a date and time', () => {
    const props: RehearsalViewComponentProps = {
      date: 'Tue, January 16',
      rehearsal: {
        time: '19:00 - 22:00',
        material: ['']
      },
      dismissRehearsal
    }
    const tree = renderer.create(
      <RehearsalViewComponent {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})