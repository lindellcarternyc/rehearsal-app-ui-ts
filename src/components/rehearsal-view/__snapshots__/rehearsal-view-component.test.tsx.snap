// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RehearsalViewComponent renders a date and time 1`] = `
<div
  style={
    Object {
      "marginTop": "3rem",
    }
  }
>
  <div
    className="ui container"
    style={
      Object {
        "marginBottom": "2rem",
        "marginTop": "3rem",
      }
    }
  >
    <h2
      className="ui header"
    >
      Tue, January 16
      <div
        className="sub header"
      >
        19:00 - 22:00
      </div>
    </h2>
    <div>
      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h4
          className="ui header"
        >
          Material
        </h4>
        <div
          className="ui bulleted list"
          role="list"
        >
          <div
            className="item"
            data-value={undefined}
            onClick={[Function]}
            role="listitem"
          >
            
          </div>
        </div>
      </div>
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Back to Day
      </button>
    </div>
  </div>
</div>
`;
